packages/web/src/routes/home.tsx

```diff
@@ -66,6 +68,29 @@ export default function Home() {
   const [appToDelete, setAppToDelete] = useState<AppType | null>(null);
   const [showCreateAppModal, setShowCreateAppModal] = useState(false);
 
+  const { aiProvider, openaiKey, anthropicKey } = useSettings();
+  const [showOnboarding, setShowOnboarding] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    const checkApiKey = () => {
+      const hasSeenOnboarding = localStorage.getItem('hasSeenOnboarding') === 'true';
```

Comment:
This should live in persistent storage (sqlite), at least in the medium-long term.

Reply:
I updated this logic to instead gate on whether `aiEnabled` being outputted from `useSettings` was enabled rather than segregating the onboarding state individually.

I don't think there's a reason that tracking this separately is important, given that this should be shown for all users that don't have an ai api key set - if so though, please call it out.